### PR TITLE
initial vector functions

### PR DESCRIPTION
**Pull request by lubankp**
@lubankp added homework vector_of_shared_ptrs

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @lubankp

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @lubankp

**Comment by ziobron** [comment]
⏱ 3 XP extra for @lubankp for on-time delivery

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
+#include "vectorFunctions.hpp"
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> vector_ptrs;
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:5]: move to function
Reserve memory when you know how many elements will be inside vector.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+std::vector<std::shared_ptr<int>> generate(int count){
+
+    for(auto i = 0; i < count; ++i){
+        vector_ptrs.push_back(std::make_shared<int>(i));
+    }
+    return vector_ptrs;
+}
+void print(std::vector<std::shared_ptr<int>> vector_ptrs){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:14]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+}
+void add10(std::vector<std::shared_ptr<int>> vector_ptrs){
+    for(auto i : vector_ptrs){
+        if(i){
+            *i += 10;
+        }
+    }
+}
+void sub10(int* const ptrs){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:26]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    }
+    return vector_ptrs;
+}
+void print(std::vector<std::shared_ptr<int>> vector_ptrs){
+    for(auto i : vector_ptrs){
+        std::cout << *i << " \n";
+    }
+}
+void add10(std::vector<std::shared_ptr<int>> vector_ptrs){
+    for(auto i : vector_ptrs){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:20]: Add space after `if`, `for`, `while` `function` etc...

Add space between operator like `+` `-` `/` `:` `&&` etc..

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp (hunk 3 above)
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:26]: Add new line between functions

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**